#!/bin/sh -e
##
## 90_debian_alsaconf.dpatch by Thomas Hood
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't precompute paths for lspci and lsmod


if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

case "$1" in
       -patch) patch $patch_opts -p1 < $0;;
       -unpatch) patch $patch_opts -p1 -R < $0;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0

@DPATCH@
diff -urNad alsa-utils-1.0.8/alsaconf/alsaconf.in /tmp/dpep.YzJVda/alsa-utils-1.0.8/alsaconf/alsaconf.in
--- alsa-utils-1.0.8/alsaconf/alsaconf.in	2005-02-18 21:40:32.000000000 +0100
+++ /tmp/dpep.YzJVda/alsa-utils-1.0.8/alsaconf/alsaconf.in	2005-02-18 21:40:32.000000000 +0100
@@ -62,12 +62,18 @@
   distribution="unknown"
 fi
 
+if false; then
 for prog in lspci lsmod; do
 	for path in /sbin /usr/sbin /bin /usr/bin;do
 		[[ -x $path/$prog ]] && eval $prog=$path/$prog
 	done
 done
 unset prog path
+else
+# debian patch
+lspci=lspci
+lsmod=lsmod
+fi
 
 usage() {
     xecho "ALSA configurator"
